### ISSUE TITLE
🔐 HTTPS Everywhere

### ISSUE THREAD

**Issue by rupl** [devsmeetup/devsmeetup.github.io#41]
![google_chrome_http](https://user-images.githubusercontent.com/254753/36197081-0f9bbfd6-1173-11e8-8770-1861ef4f1cfa.png)

**Whoever controls DNS of http://devsmeetup.de needs to perform this task.** CloudFlare lets you do this for free with gh-pages.

1. Sign up for CloudFlare.
2. Add Site and import DNS records.
3. Point domain at CloudFlare nameservers.
4. Enable **Full SSL** under **Crypto** settings:

<img width="978" alt="cloudflare-crypto-ssl" src="https://user-images.githubusercontent.com/254753/36197481-21b7385c-1174-11e8-8f97-dab4af9ebad2.png">

5. Set up a **Page Rule** to **Always use HTTPS** when users attempt to hit the non-secure URL. The rule should be written for this exact pattern: `http://*devsmeetup.de/*`

<img width="984" alt="cloudflare-pagerule-https" src="https://user-images.githubusercontent.com/254753/36197491-2bd16ee8-1174-11e8-8442-b272e186d3ab.png">

_You might need to wait for SSL to activate before doing step 5, but sometimes it's fast enough that you can do it all at once._

6. Present this process at the meetup so everyone else can prepare for July 😜

**Comment by sebald** [comment]
Thx! I totally overlooked your issue. We actually host directly via GitHub pages + CNAME. But don't have control over the domain.

... But I'll try to enable HTTPS tomorrow. Maybe this is enough 😀

**Comment by sebald** [comment]
Note for future me: https://blog.github.com/2018-05-01-github-pages-custom-domains-https/

**Comment by rupl** [comment]
Yeah as of May it's provided for free via GitHub Pages. I don't believe any extra effort is required if you used a `CNAME`, but [an `A` record requires new IPs](https://help.github.com/articles/setting-up-an-apex-domain/#configuring-a-records-with-your-dns-provider).

**Comment by sebald** [comment]
I checked the settings, sadly I can not just enable it 😔

<img width="699" alt="screenshot 2018-05-25 00 39 02" src="https://user-images.githubusercontent.com/985701/40517146-139fef50-5fb4-11e8-8771-f6a15c6dfb5e.png">

**Comment by rupl** [comment]
When I `dig` devsmeetup.de I see the legacy IP. Checking DNS for devsmeetup.github.io points to the newer HTTPS-enabled IPs. I think this mismatch is the problem. Just ping whomever controls DNS and once it switches over it will probably upgrade automatically.

```
$ dig devsmeetup.de +nostats +nocomments +nocmd

; <<>> DiG 9.10.6 <<>> devsmeetup.de +nostats +nocomments +nocmd
;; global options: +cmd
;devsmeetup.de.			IN	A
devsmeetup.de.		86400	IN	A	192.30.252.153
```

```
$ dig devsmeetup.github.io +nostats +nocomments +nocmd

; <<>> DiG 9.10.6 <<>> devsmeetup.github.io +nostats +nocomments +nocmd
;; global options: +cmd
;devsmeetup.github.io.		IN	A
devsmeetup.github.io.	3600	IN	CNAME	sni.github.map.fastly.net.
sni.github.map.fastly.net. 1184	IN	A	185.199.111.153
sni.github.map.fastly.net. 1184	IN	A	185.199.110.153
sni.github.map.fastly.net. 1184	IN	A	185.199.108.153
sni.github.map.fastly.net. 1184	IN	A	185.199.109.153
```

**Comment by sebald** [comment]
I see words, but I don't understand them 😇

@nimius-nicki Hi Nicki, ihr seid doch so lieb und hostet unsere Meetup Seite. Hättest du Zeit HTTPS zu aktivieren? 🙂

**Comment by nicodh** [comment]
HTTPS is enabled now for devsmeetup.de but the certificate is still missing so https://devsmeetup.de is "unsafe"...

**Comment by sebald** [comment]
I still can't check the "enforce HTTPS" box in the settings. @rupl is the SSL certificate provided by Github or the hoster. @nicodh  If the later is true, did Nimius add a "Lets Encrypt" cert or something? :)

**Comment by sebald** [comment]
I think I get @rupl 's comment now. We have to use Cloudflare as a proxy. Correct?

**Comment by rupl** [comment]
Until a few weeks ago using CloudFlare was the easiest, cheapest method I knew of to use LetsEncrypt and get a free cert, hence the original issue instructions.

As of May 2018, GitHub provides the LetsEncrypt cert automatically, provided you have the CNAME set correctly (either in the repo's settings or a `CNAME` file in the branch). I do see your `CNAME` file so I'm also a little puzzled.

```
$ dig devsmeetup.de +nostats +nocomments +nocmd

; <<>> DiG 9.10.6 <<>> devsmeetup.de +nostats +nocomments +nocmd
;; global options: +cmd
;devsmeetup.de.			IN	A
devsmeetup.de.		86400	IN	A	185.199.111.153
devsmeetup.de.		86399	IN	NS	dns.typo3server.com.
devsmeetup.de.		86399	IN	NS	ns3.dnsmittwald.de.
devsmeetup.de.		86399	IN	NS	ns2.typo3server.com.
dns.typo3server.com.	7200	IN	A	81.3.28.186
ns2.typo3server.com.	7200	IN	A	188.94.252.153
ns3.dnsmittwald.de.	86400	IN	A	188.94.251.53
```

The DNS has changed since my previous comment, and shows the block of info above. The first `A` record is one of the GitHub IPs, so I'm not sure what's wrong now.  I have rarely used A records to point to GitHub pages so I'll have to inspect more deeply when I have more time. I'm actually about to board a flight back to :de: but I'll be able to look into it this week once I'm settled.

**Comment by rupl** [comment]
Ok actually I think I got it. Step 4 of [setting A records on GitHub Pages](https://help.github.com/articles/setting-up-an-apex-domain/) (emphasis mine):

> Add your custom domain to your GitHub Pages site. If you're updating the IP address of an existing A record, first remove and then re-add your custom domain to the repository you’re using to publish your Pages site to trigger the process of enabling HTTPS. For more information, see "Adding or removing a custom domain for your GitHub Pages site."

Since the previous A record was the old GH Pages IP I think you can just remove the custom domain, save settings, and add it again. Maybe that will work?

**Comment by sebald** [comment]
Hm, I tried that already, but maybe the second time is the charm!? 🙂

BTW, sick UX. If you delete the custom domain, GitHub will automatically delete the CNAME for you! (https://github.com/devsmeetup/devsmeetup.github.io/commit/3c137bd8d2bdf4c5268921e281ab29e81cc0a708)

Some observations:

1. After deleting the custom domain, http://devsmeetup.github.io/ still points to devsmeetup.de (so is the https version)
1. I would like to completely unpublish the site, but since it is called "devsmeetup.github.io" this doesn't work 

I will wait 15min or so and re-add the custom domain ... hopefully it will work this time.

**Comment by sebald** [comment]
![https://media.giphy.com/media/GIEXgLDfghUSQ/giphy.gif](https://media.giphy.com/media/GIEXgLDfghUSQ/giphy.gif)

<img width="765" alt="screenshot 2018-05-29 13 31 37" src="https://user-images.githubusercontent.com/985701/40656430-a2dfd3d6-6344-11e8-8f11-aec8343a24f1.png">

... at least I can now check the box! \o/ Hopefully it is working when the DNS is updated.

**Comment by sebald** [comment]
Well ...

<img width="922" alt="screenshot 2018-05-29 13 55 42" src="https://user-images.githubusercontent.com/985701/40657415-134f4afe-6348-11e8-8c37-fb958f2f5c51.png">

Can some one else please try and confirm it is working?

**Comment by dignifiedquire** [comment]
fine for me

**Comment by nicodh** [comment]
Sieht gut aus!!
[
<img width="321" alt="bildschirmfoto 2018-05-29 um 14 00 26" src="https://user-images.githubusercontent.com/1428457/40657596-bac51638-6348-11e8-94bc-ec72acd5a668.png">
](url)

**Comment by rupl** [comment]
all good for me too!  🎉
